pkg/chains/signing/kms/kms.go

```diff
@@ -95,12 +98,27 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
 	// pass through configuration options to RPCAuth used by KMS in sigstore
 	rpcAuth := options.RPCAuth{
 		Address: cfg.Auth.Address,
-		Token:   cfg.Auth.Token,
+		//Token:   cfg.Auth.Token,
 		OIDC: options.RPCAuthOIDC{
 			Role: cfg.Auth.OIDC.Role,
 			Path: cfg.Auth.OIDC.Path,
 		},
 	}
+
+	// get token from file KMS_AUTH_TOKEN, a mounted secret at signers.kms.auth.token-dir or
+	// as direct value set from signers.kms.auth.token.
+	// If both values are set, priority will be given to token-dir.
+
+	if cfg.Auth.TokenDir != "" {
+		rpcAuthToken, err := getRPCAuthToken(cfg.Auth.TokenDir)
```

Comment:
If we are adding for all kms providers 

```suggestion
		rpcAuthToken, err := getKMSAuthToken(cfg.Auth.TokenDir)
```

Reply:
changed